spec/controllers/restaurants_controller_spec.rb

```diff
@@ -0,0 +1,104 @@
+require 'rails_helper'
+
+RSpec.describe RestaurantsController, type: :controller do
+  describe "actions" do
+    describe "GET index" do
+      let!(:restaurants) do
+        FactoryBot.create_list(:restaurant, 2)
+      end
+
+      before(:each) do
+        get(:index)
+      end
+
+      it "checking count" do
+        expect(JSON.parse(response.body).size).to eq(2)
+      end
+
+      it "checking creation status" do
+        expect(response).to have_http_status(:success)
+      end
+    end
+
+    describe "GET show" do
+      let!(:restaurant) do
+        FactoryBot.create(:restaurant)
+      end
+
+      it "checking show" do
+        get(:show, params: {:id => restaurant[:id]})
+        expect(JSON.parse(response.body)).to eq(YAML.load(RestaurantSerializer.new(restaurant).to_json))
+      end
+    end
+
+    describe "POST create" do
+      before(:each) do
```

Comment:
in this test since you are creating an entity, may be worth to have an `after` action do clear the created restaurant so tests won't break or be flaky.

Reply:
I don't think it is necessary since RSpec works in a way that data that is being modified in the 'before' hook, will be rollback after the example running.